prereise/gather/demanddata/bldg_electrification/load_projection.py

```diff
@@ -0,0 +1,674 @@
+
+    heat_eng = 0
+    mid_cool_eng = 0
+    cool_eng = 0
+    hp_eng = 0
+    resist_eng = 0
+
+    wk_wknd = (
+        "wk"
+        if temp_series["weekday"] < 5 and ~temp_series["holiday"]  # boolean value
+        else "wknd"
+    )
+
+    (
+        t_bpc,
+        t_bph,
+        i_heat,
+        s_heat,
+        s_dark,
+        i_cool,
+        s_cool_db,
+        s_cool_wb,
+    ) = (
+        hourly_fits_df.at[zone_hour, f"t.bpc.{wk_wknd}.c"],
+        hourly_fits_df.at[zone_hour, f"t.bph.{wk_wknd}.c"],
+        hourly_fits_df.at[zone_hour, f"i.heat.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"s.heat.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"s.dark.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"i.cool.{wk_wknd}"],
+        hourly_fits_df.at[zone_hour, f"s.cool.{wk_wknd}.db"],
+        hourly_fits_df.at[zone_hour, f"s.cool.{wk_wknd}.wb"],
+    )
+
+    base_eng = s_heat * t_bph + s_dark * dark_frac + i_heat
+
+    if temp <= t_bph:
+        heat_eng = -s_heat * (t_bph - temp)
+        # seperate resistance heat and heat pump energy by COP. HP assumed to be midium performance HP that's used in Mike's Joule paper
+        # ! Since we now have our estimation of current heat pump adoption rate, this function can be merged to the same one in zone_profile_generator.py
+        cop_hp = hp_heat_cop.loc[round(temp, 1), "cop"]
+        hp_eng = (
+            heat_eng
+            * (base_scen.hp_heat_frac / cop_hp)
+            / (base_scen.resist_heat_frac + base_scen.hp_heat_frac / cop_hp)
+        )
+        resist_eng = (
+            heat_eng
+            * (base_scen.resist_heat_frac)
+            / (base_scen.resist_heat_frac + base_scen.hp_heat_frac / cop_hp)
+        )
+
+    if temp >= t_bph:
+        cool_eng = (
+            s_cool_db * temp
+            + s_cool_wb
+            * (
+                temp_wb
+                - (db_wb_fit[0] * temp**2 + db_wb_fit[1] * temp + db_wb_fit[2])
+            )
+            + i_cool
+        )
+
+    if temp > t_bpc and temp < t_bph:
+        mid_cool_eng = ((temp - t_bpc) / (t_bph - t_bpc)) ** 2 * (
+            s_cool_db * t_bph
+            + s_cool_wb
+            * (
+                temp_wb
+                - (db_wb_fit[0] * temp**2 + db_wb_fit[1] * temp + db_wb_fit[2])
+            )
+            + i_cool
+        )
+
+    return [base_eng, hp_eng, resist_eng, max(cool_eng, 0) + max(mid_cool_eng, 0)]
+
+
+def scale_energy(
+    base_energy, temp_df, base_scen, new_scen, midperfhp_cop, advperfhp_cop
+):
+    """project energy consumption for each projection scenarios from the base scenario
+
+    :param pandas.DataFrame base_energy: dataframe of disaggregated electricity consumptions for all weather years
+    :param pandas.DataFrame temp_df: weather records the given hours
+
+    :return (*pandas.DataFrame*) scen_load_mwh -- hourly electricity consumption induced by heat pump heating, resistance heating, cooling, and 'base' loads for a projection scenario
```

Comment:
Doc strings need to be consistent with the parameters.

Reply:
Done